tools/CleanupBuild.ps1

```diff
@@ -5,17 +5,40 @@ Param
     [string]$BuildConfig
 )
 
-function Cleanup-MarkdownHelp
-{
-    [CmdletBinding()]
-    Param
-    (
-        [Parameter()]
-        [string]$BuildConfig
-    )
+$output = Join-Path (Get-Item $PSScriptRoot).Parent.FullName "src\Package\$BuildConfig"
+Write-Verbose "The output folder is set to $output"
+$serviceManagementPath = Join-Path $output "ServiceManagement\Azure"
+$resourceManagerPath = Join-Path $output "ResourceManager\AzureResourceManager"
+
+Write-Verbose "Removing generated NuGet folders from $output"
+$resourcesFolders = @("de", "es", "fr", "it", "ja", "ko", "ru", "zh-Hans", "zh-Hant")
+Get-ChildItem -Include $resourcesFolders -Recurse -Force -Path $output | Remove-Item -Force -Recurse -ErrorAction SilentlyContinue
+
+Write-Verbose "Removing XML help files for helper dlls from $output"
+$exclude = @("*.dll-Help.xml", "Scaffold.xml", "RoleSettings.xml", "WebRole.xml", "WorkerRole.xml")
+$include = @("*.xml", "*.lastcodeanalysissucceeded", "*.dll.config", "*.pdb")
+Get-ChildItem -Include $include -Exclude $exclude -Recurse -Path $output | Remove-Item -Force -Recurse
+Get-ChildItem -Recurse -Path $output -Include *.dll-Help.psd1 | Remove-Item -Force
 
-    $HelpFolders = Get-ChildItem "help" -Recurse -Directory | where { $_.FullName -like "*$BuildConfig*" }
-    $HelpFolders | foreach { Remove-Item -Path $_ -Recurse -Force }
-}
+Write-Verbose "Removing markdown help files and folders"
+Get-ChildItem -Recurse -Path $output -Include *.md | Remove-Item -Force
+Get-ChildItem -Directory -Include help -Recurse -Path $output | Remove-Item -Force
+
+Write-Verbose "Removing unneeded web deployment dependencies"
+$webdependencies = @("Microsoft.Web.Hosting.dll", "Microsoft.Web.Delegation.dll", "Microsoft.Web.Administration.dll", "Microsoft.Web.Deployment.Tracing.dll")
+Get-ChildItem -Include $webdependencies -Recurse -Path $output | Remove-Item -Force
+
+$resourceManagerFolders = Get-ChildItem -Path $resourceManagerPath -Directory
+foreach ($RMFolder in $resourceManagerFolders)
+{
+    $psd1 = Get-ChildItem -Path $RMFolder.FullName -Filter "$($RMFolder.Name).psd1"
+    Import-LocalizedData -BindingVariable ModuleMetadata -BaseDirectory $psd1.DirectoryName -FileName $psd1.Name
+    
+    $acceptedDlls = @()
+    $acceptedDlls += $ModuleMetadata.NestedModules
+    $acceptedDlls += $ModuleMetadata.RequiredAssemblies
 
-Cleanup-MarkdownHelp -BuildConfig $BuildConfig
\ No newline at end of file
+    $acceptedDlls = $acceptedDlls | where { $_ -ne $null } | % { $_.Substring(2) }
+    
+    Get-ChildItem -Path $RMFolder.FullName -Filter "*.dll" | where { $acceptedDlls -notcontains $_.Name} | Remove-Item -Force
```

Comment:
It might be worthwhile to ensure we print out these files we are removing so we can debug any issues